### PR TITLE
Fix AnyVal Pickler derivation

### PR DESCRIPTION
**Pull request by notxcain**
I don't know why existing tests compile. But outside of the project derived `Pickler` for `AnyVal` fails to compile. Because `identityRefFor` accepts `AnyRef`.

Also I bumped `circe-jawn` because there is no `0.2.1` published for `Scala 2.12.x`.

### CODE REVIEW

**Comment by cornerman** [comment]
Interesting, I am using AnyVals in a project of mine and works fine (just like the tests). Can you provide an example where it does not work?

File: boopickle/shared/src/main/scala/boopickle/PicklerMaterializersImpl.scala [hunk 1]
```diff
@@ -117,7 +117,10 @@ object PicklerMaterializersImpl {
         q"""state.pickle[$fieldTpe](value.${accessor.name})"""
       }
 
-      q"""
+      if (sym.isDerivedValueClass)
```
Comment (cornerman) [inline boopickle/shared/src/main/scala/boopickle/PicklerMaterializersImpl.scala]: Why is this needed? We have [a check for this a few lines before](https://github.com/notxcain/boopickle/blob/d57ea13098e358d7f4a912e78cc20a0a627dec15/boopickle/shared/src/main/scala/boopickle/PicklerMaterializersImpl.scala#L95). How can we reach this point?

Comment (notxcain) [reply boopickle/shared/src/main/scala/boopickle/PicklerMaterializersImpl.scala]: That's strange indeed! I didn't notice this case.

Could you please take a look at https://github.com/notxcain/boopickle-repro, it fails to compile without this fix.

Comment (notxcain) [reply boopickle/shared/src/main/scala/boopickle/PicklerMaterializersImpl.scala]: That's strange but if I `publishLocal` from `master` my reproducer compiles. But it fails to compile with `1.2.6`

Comment (cornerman) [reply boopickle/shared/src/main/scala/boopickle/PicklerMaterializersImpl.scala]: Ah! This is not released yet in 1.2.6.

For now you can get it via jitpack:
```scala
resolvers += "jitpack" at "https://jitpack.io"
libraryDependencies += "com.github.suzaku-io.boopickle" %% "boopickle" % "master-SNAPSHOT"
```

**Comment by notxcain** [comment]
@cornerman okay then. I'll wait for release and will submit a separate PR updating `circe-jawn` to `0.9.2`.
